airflow/models/taskinstance.py

```diff
@@ -400,6 +400,13 @@ def key(self) -> "TaskInstanceKey":
         return self
 
 
+def executor_config_comparator(x, y):
+    try:
+        return x == y
+    except AttributeError:
```

Comment:
Wait, how can this raise AttributeError?

Reply:
cus it blows up when you call to_dict and that's how `__eq__` is defined:
```python
    def __eq__(self, other):
        """Returns true if both objects are equal"""
        if not isinstance(other, V1Container):
            return False

        return self.to_dict() == other.to_dict()
```

if you look at the to_dict code in on a kube object in the k8s library, it relies on on the current api structure and calls getattr for each attr that should be there, and if version not compatble, it will try to access an attr that isn't there.

it's confusing... trying to see what we can do about this.

Reply:
Ah, yes, that makes sense, thanks. Didn't think of the version compatibility aspect of `__eq__`.

Reply:
Instead of catching AttributeError here (which is confusing), can this failover happen in `__eq__` instead?

Reply:
yeah if we were using our own PickleType then we'd be able to do that.  but we're using sqlalchemy's PickleType.  Seems that it's not worth subclassing when the class already provides a mechanism (i.e. compator) for handling this kind of thing.  WDYT?

Reply:
Is the exception raised from PickleType? In that case I guess a comment here is the best we can do.